removed dependency on bart tokenizer(slow/fast) LED

Infrared1029 opened the pull request:
# What does this PR do?

removes the dependency between LED tokenizer and Bart's (slow version)

Fixes # (issue)

follows hugging's face philosophy of Do Repeat Yourself


## Who can review?
@sgugger

Infrared1029 committed:
removed dependency from bart(slow)

Infrared1029 committed:
removed dependency from bart(slow)

Infrared1029 reviewed (commented):
~The docstring where it shows how to use the tokenizer, what LED model goes there? (ive put a comment there asking)~, nvm found it

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_19388). All of your documentation changes will be reflected on that endpoint.

Ilaygoldman committed:
Change link of repojacking vulnerable link (#19393)

The link to https://github.com/vasudevgupta7/bigbird is vulnerable to repojacking (it redirects to the orignial project that changed name), you should change the link to the current name of the project. if you won't change the link, an attacker can open the linked repository and attacks users that trust your links

alaradirik committed:
fix docs example, add object_detection to DETR docs (#19377)

Infrared1029 committed:
adding copying comments (copied from bart to led)

Infrared1029 committed:
added comments for copied functions (bard to led)

Infrared1029 committed:
updated led docstring

Infrared1029 committed:
updated led docstring

Infrared1029 committed:
removed dependency from Bart (fast)

Infrared1029 committed:
replaced bart with LED in docstrings

Infrared1029 committed:
Merge branch 'ind_models' of github.com:Infrared1029/transformers into ind_models

Infrared1029 committed:
complying flake8

Infrared1029 committed:
added more copy comments

sgugger commented on src/transformers/models/led/tokenization_led.py:
```diff
@@ -40,18 +49,378 @@
+    return pairs
+
+
+# Copied from transformers.models.bart.tokenization_bart.BartTokenizer with Bart->LED
```
```suggestion
# Copied from transformers.models.bart.tokenization_bart.BartTokenizer with Bart->LED,facebook/bart-base->allenai/led-base-16384
```
We also need to update the checkpoint name here.

sgugger commented on src/transformers/models/led/tokenization_led.py:
```diff
@@ -40,18 +49,378 @@
     max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.__init__
```
```suggestion
```
You have the global copied from on the class, so you won't need this one.

sgugger commented on src/transformers/models/led/tokenization_led.py:
```diff
@@ -40,18 +49,378 @@
+        self.pat = re.compile(r"""'s|'t|'re|'ve|'m|'ll|'d| ?\p{L}+| ?\p{N}+| ?[^\s\p{L}\p{N}]+|\s+(?!\S)|\s+""")
+
+    @property
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.vocab_size
+    def vocab_size(self):
+        return len(self.encoder)
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.get_vocab
+    def get_vocab(self):
+        return dict(self.encoder, **self.added_tokens_encoder)
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.bpe
```
Same for those and all methods below.

sgugger commented on src/transformers/models/led/tokenization_led_fast.py:
```diff
@@ -41,19 +47,251 @@
 
 
-class LEDTokenizerFast(BartTokenizerFast):
+# Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast with Bart->LED
```
```suggestion
# Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast with Bart->LED,facebook/bart-base->allenai/led-base-1638
```

sgugger commented on src/transformers/models/led/tokenization_led_fast.py:
```diff
@@ -41,19 +47,251 @@
     slow_tokenizer_class = LEDTokenizer
+    model_input_names = ["input_ids", "attention_mask"]
+
+    # Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast.__init__
```
And same here: no need for other Copied from statements.

sgugger reviewed (commented):
Thanks a lot for your PR! I've left a couple of comments to fix some of your Copied from statements.

With your rebase, it looks like the PR contains modifications that should not be there. I think the easiest fix for that is to close this one and open a fresh one from the same branch :-)

Infrared1029 commented:
Thanks a lot for the feedback! ill fix everything and open a brand new one.
